src/Application/Features/Genres/Commands/DeleteGenre.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Carter;
+using LyricsApp.Application.Common.Models;
+using LyricsApp.Application.Domain.Entities;
+using LyricsApp.Application.Infrastructure.Persistence;
+using MediatR;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.EntityFrameworkCore;
+
+
+namespace LyricsApp.Application.Features.Genres.Commands
+{
+    public class DeleteGenre : ICarterModule
+    {
+        public void AddRoutes(IEndpointRouteBuilder app)
+        {
+            app.MapDelete("api/genre/{genreId}", async (IMediator mediator, string genreId) =>
+            {
+                return await mediator.Send(new DeleteGenreCommand(genreId));
+            })
+            .WithName(nameof(DeleteGenre))
+            .WithTags(nameof(Genre))
+            .Produces(StatusCodes.Status200OK, typeof(BasicResponse<>))
+            .Produces(StatusCodes.Status500InternalServerError)
+            .Produces(StatusCodes.Status404NotFound, typeof(BasicResponse<>));
+        }
+    }
+
+    public class DeleteGenreCommand : IRequest<IResult>
+    {
+        public DeleteGenreCommand(string genreId) => GenreId = genreId;
+
+        public string GenreId { get; set; }
```

Comment:
```suggestion
        public Guid GenreId { get; set; }
```
Puedes usar Guid desde aqui